resources/run_adl.sh

```diff
@@ -0,0 +1,195 @@
+MONGO_DB_PATH=$(pwd)/test_db
+LOGS_PATH=$(pwd)/logs
+MONGOD_PORT=28017
+MONGOHOUSED_PORT=27017
+START="start"
+STOP="stop"
+MONGOD="mongod"
+MONGOHOUSED="mongohoused"
+MONGO_DOWNLOAD_LINK=
+OS=$(uname)
+
+MONGO_DOWNLOAD_BASE=https://fastdl.mongodb.org
+# Ubuntu 18.04
+MONGO_DOWNLOAD_UBUNTU=mongodb-linux-x86_64-ubuntu1804-5.0.4.tgz
+# RedHat 7
+MONGO_DOWNLOAD_REDHAT=mongodb-linux-x86_64-rhel70-5.0.4.tgz
+# macOS
+MONGO_DOWNLOAD_MAC=mongodb-macos-x86_64-5.0.4.tgz
+
+check_procname() {
+  ps -ef 2>/dev/null | grep $1 | grep -v grep >/dev/null 
+  result=$?
+  
+  if [[ result -eq 0 ]]; then
+    return 0
+  else 
+    return 1
+  fi
+}
+
+check_port() {
+  netstat -van 2>/dev/null | grep LISTEN | grep $1 >/dev/null
+  result=$?
+  
+  if [[ result -eq 0 ]]; then
+    return 0
+  else 
+    return 1
+  fi
+}
+
+check_mongod() {
+  check_procname $MONGOD
+  process_check_result=$?
+  check_port $MONGOD_PORT
+  port_check_result=$?
+
+  if [[ $process_check_result -eq 0 ]] && [[ $port_check_result -eq 0 ]]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+check_mongohoused() {
+  check_procname $MONGOHOUSED
+  process_check_result=$?
+  check_port $MONGOHOUSED_PORT
+  port_check_result=$?
+
+  if [[ $process_check_result -eq 0 ]] && [[ $port_check_result -eq 0 ]]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+# check if mac or linux
+if [ $OS = "Linux" ]; then
+  distro=$(awk -F= '/^NAME/{print $2}' /etc/os-release)
+  if [ "$distro" = "\"Red Hat Enterprise Linux\"" ] ||
+[ "$distro" = "\"Red Hat Enterprise Linux Server\"" ]; then
+    export VARIANT=rhel7
+    MONGO_DOWNLOAD_LINK=$MONGO_DOWNLOAD_REDHAT
+  elif [ "$distro" = "\"Ubuntu\"" ]; then
+    export VARIANT=ubuntu1804  
+    MONGO_DOWNLOAD_LINK=$MONGO_DOWNLOAD_UBUNTU
+  else
+    echo ${distro} not supported
+    exit 1
+  fi
+elif [ $OS = "Darwin" ]; then
+  MONGO_DOWNLOAD_LINK=$MONGO_DOWNLOAD_MAC
+else
+  echo $(uname) not supported
+  exit 1
+fi
+
+check_mongod
+if [[ $? -ne 0 ]]; then
+  if [ $ARG = $START ]; then
+    # Install and start mongod
+    if [ $OS = "Linux" ]; then
+      curl -O $MONGO_DOWNLOAD_BASE/linux/$MONGO_DOWNLOAD_LINK
+    else
+      curl -O $MONGO_DOWNLOAD_BASE/osx/$MONGO_DOWNLOAD_LINK
+    fi
+
+    tar zxvf $MONGO_DOWNLOAD_LINK
+
+    mkdir -p $MONGO_DB_PATH
+    mkdir -p $LOGS_PATH
+    mkdir -p $TMP_DIR
+
+    # Note: ADL has a storage.json file that generates configs for us.
+    # The mongodb source is on port $MONGOD_PORT so we use that here.
+    ${MONGO_DOWNLOAD_LINK:0:$((${#MONGO_DOWNLOAD_LINK} - 4))}/bin/mongod --port $MONGOD_PORT --dbpath $MONGO_DB_PATH \
+      --logpath $LOGS_PATH/mongodb_test.log --pidfilepath $TMP_DIR/${MONGOD}.pid --fork
+  fi
+else
+  if [ $ARG = $STOP ]; then
+    MONGOD_PID=$(< $TMP_DIR/${MONGOD}.pid)
+    echo "Stopping $MONGOD, pid $MONGOD_PID"
+    kill "$(< $TMP_DIR/${MONGOD}.pid)"
+  fi
+fi
+
+check_mongohoused
+if [[ $? -ne 0 ]]; then
+  if [ $ARG = $START ]; then
+    # Install and start mongohoused
```

Comment:
Let's consider adding` echo "Starting mongohoused"` like we have logging for stopping it in the else branch.